### ISSUE TITLE
WPF App not sending unhandled exception in Release build

### ISSUE THREAD

**Issue by sebaJoSt** [microsoft/appcenter-sdk-dotnet#1727]
I have created a new WPF Application (C#, Windows, Desktop) in .Net 7.0. Added a button with the following code:
```
        private void Button_Click(object sender, RoutedEventArgs e)
        {
#if DEBUG
            throw new Exception("Debug build: Intentionally crashed the app");
#else
            throw new Exception("Release build: Intentionally crashed the app");
#endif
        }
```
In debug build the crash is successfully sent to App Center after restarting the app. In release build nothing happens.

Here's the additional code in App.xaml.cs:
```
   protected async override void OnStartup(StartupEventArgs e)
        {
            base.OnStartup(e);

            AppCenter.Start("Your app secret here",
                  typeof(Analytics), typeof(Crashes));
        }
```

I removed the app secret in this problem report.

Details
Platform: Win10, WPF. .NET 7.0
SDK version 5.0.1
OS: Windows 10
Visual Studio Version 17.4.4

**Comment by aleksandr-dorofeev** [comment]
Hello @sebaJoSt thanks for getting in touch with us!
Using the steps you provided, I was unable to reproduce the behavior you describe. Sending a crash report works in both cases on my end.
Could you provide a sample app with the problem described?

**Comment by sebaJoSt** [comment]
Hello @aleksandr-dorofeev , thanks for looking into this.

Here's my example repository: 
https://github.com/sebaJoSt/CrashTestApp

![image](https://user-images.githubusercontent.com/388152/214825571-3f3fcd00-395d-40e9-b371-e1a51d482816.png)
As you can see in this screenshot, only the debug crash appears.

**Comment by aleksandr-dorofeev** [comment]
Thanks for providing the app. 
I tested sending crash in the release build on your sample app and it was sent successfully and then visible on the portal.
Please turn on [verbose logs](https://learn.microsoft.com/en-gb/appcenter/sdk/other-apis/wpf-winforms#adjust-the-log-level) and make sure that you can see the crash payload in logs and the http status code of sending it is 200. In that case the SDK works correctly, and if you can't see the log on the portal for more than 10 minutes, please contact support team.

**Comment by sebaJoSt** [comment]
I turned on verbose logs and i see everything working in debug build. But in release build, where it doesn't work, I can't see the logs, can the output be written in a file?

I still think its a problem with the sdk, because in the InstallId folders the crashes are added in debug build, but not in release.
Here are the code changes:
```
  protected async override void OnStartup(StartupEventArgs e)
        {
            base.OnStartup(e);
            AppCenter.LogLevel = LogLevel.Verbose;
            AppCenter.Start("42466295-5bcc-4139-9c40-81e056746615",
                  typeof(Analytics), typeof(Crashes));

            System.Guid? installId = await AppCenter.GetInstallIdAsync();
            await File.WriteAllTextAsync("C:\\Users\\stroblse\\Desktop\\InstallId.txt", installId.ToString());

        }
```

### Release screen capture
No EXCEPTION and no JSON File added
![release_mode](https://user-images.githubusercontent.com/388152/215452266-ab0ca78f-8508-4ed5-a494-c3f10973476a.gif)

### Debug screen capture
EXCEPTION and JSON File are added -> Next time I start the app, the files are deleted and the crash be found in the AppCenter
![debug_mode](https://user-images.githubusercontent.com/388152/215452318-285da7fd-74bf-44b5-8301-677770668293.gif)

**Comment by sebaJoSt** [comment]
any news on this topic? can you reproduce this problem?

**Comment by MikhailSuendukov** [comment]
@sebaJoSt I was able to reproduce this error, so I created a bug on our dashboard.

**Comment by sebaJoSt** [comment]
It's about two month and this is pretty essential functionality not working. Any progress here?

**Comment by MikhailSuendukov** [comment]
Hello, we are currently polishing a fix for this bug [here](https://github.com/microsoft/appcenter-sdk-dotnet/pull/1737). While we are doing this, I can offer you a workaround for submitting crashes to the `AppCenter` portal. You will need to add this code to your Application class:

```
public partial class App : Application
{
        protected override void OnStartup(StartupEventArgs e)
        {
            base.OnStartup(e);
            AppCenter.Start("secret",
                  typeof(Analytics), typeof(Crashes));
            DispatcherUnhandledException += App_DispatcherUnhandledException;
        }

        void App_DispatcherUnhandledException(object sender, System.Windows.Threading.DispatcherUnhandledExceptionEventArgs e)
        {
            Crashes.TrackError(e.Exception);
        }
}
```

**Comment by snakefoot** [comment]
Resolved by AppCenter ver 5.0.2 ? Or still waiting for #1742 ?

See also: https://github.com/microsoft/appcenter-sdk-dotnet/blob/develop/CHANGELOG.md

**Comment by MikhailSuendukov** [comment]
Yes, the fix was released in the latest version. I am closing this issue, but if you keep getting the error or have some questions, feel free to re-open this issue.
